splash/browser_tab.py

```diff
@@ -870,6 +870,23 @@ def _post_mouse_event(self, type, button, x, y):
         event = QMouseEvent(type, point, q_button, buttons, modifiers)
         QApplication.postEvent(self.web_page, event)
 
+    def send_keys(self, text, key_type=None):
+        """
+        Send key events to webpage
+        :param text string to be sent as key strokes
+        :param key_type string representing the pressed key
+        :return: None
+        """
+        key_type = key_type or 'unknown'
+        # http://doc.qt.io/qt-5/qt.html#Key-enum
+        key_type = getattr(Qt, 'Key_%s' % key_type, Qt.Key_unknown)
+        self._send_keys(text, key_type)
+
+    def _send_keys(self, text, key_type):
+        modifiers = QApplication.keyboardModifiers()
+        event = QKeyEvent(QEvent.KeyPress, key_type, modifiers, text)
```

Comment:
I saw no apparent difference when sending a KeyRelease after KeyPress, so decided to go without it.

I see in this QT example they send an empty KeyRelease event, though it's for single keys.
```C++
void MyInputPanelContext::sendCharacter(QChar character)
{
    QPointer<QWidget> w = focusWidget();

    if (!w)
        return;

    QKeyEvent keyPress(QEvent::KeyPress, character.unicode(), Qt::NoModifier, QString(character));
    QApplication::sendEvent(w, &keyPress);

    if (!w)
        return;

    QKeyEvent keyRelease(QEvent::KeyPress, character.unicode(), Qt::NoModifier, QString());
    QApplication::sendEvent(w, &keyRelease);
}

```
http://doc.qt.io/qt-4.8/qt-tools-inputpanel-example.html

To note on this example, they send as key code the unicode value of the character. Might work, if we are to implement keyboard input on a loop.